<?xml version="1.0" encoding="utf-8"?>

<launch>
  
  <arg name="launch_prefix" default="xterm -e"/>

  <!-- Config and weights folder -->
  <arg name="yolo_weights_path"          default="$(find cloth_segmentation)/yolo_network_config/weights"/>
  <arg name="yolo_config_path"           default="$(find cloth_segmentation)/yolo_network_config/cfg"/>

  <!-- ROS and network parameter files -->
  <arg name="ros_param_file"             default="$(find darknet_ros)/config/ros.yaml"/>
  <arg name="network_param_file"         default="$(find cloth_segmentation)/config/yolo_cloth.yaml"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="darknet_ros" file="$(arg ros_param_file)"/>
  <rosparam command="load" ns="darknet_ros" file="$(arg network_param_file)"/>
  <rosparam command="load" ns="darknet_ros" file="$(find cloth_segmentation)/config/params.yaml"/>

  <!-- Start darknet and ros wrapper -->
  <node pkg="darknet_ros" type="darknet_ros" name="darknet_ros" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="weights_path"          value="$(arg yolo_weights_path)" />
    <param name="config_path"           value="$(arg yolo_config_path)" />
  </node>

</launch>
